app/controllers/users/settings/user_settings_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Users
+  module Settings
+    class UserSettingsController < ApplicationController
+      def show
+        render json: { data: current_user.settings[params[:key]] }
+      end
+
+      def update
+        key = params[:key]
+        data = params[:data].permit!
```

Comment:
You can just use normal rails params with key `settings` and just `merge`. In future then it will be possible to save multiple keys in same time.